### DISCUSSION TITLE
Can I get original value of PyErr?

### DISCUSSION THREAD

**Discussion by bachue** [PyO3/pyo3#2378] [Questions] [answered]
I just define a `pyclass` struct to store a rust error;

```
#[pyclass]
pub struct IoError(std::io::Error);
```

and pass its instance into `PyIOError::new_err(IoError(...))`

How can I get the original `std::io::Error` from the `PyIOError` or `PyErr`? 
I tried to call `PyErr::value()`, but still cannot extract `IoError` from it.
Thanks

**Comment by birkenfeld** [comment, marked as answer]
Your IoError is not an exception. The value is a standard Python IOError, but you can get at the "contents" with (in Python) `err.args[0]`.

**Reply by bachue** [reply]
@birkenfeld Thanks for your reply, but can I get the value of `err.args[0]` in Rust?

**Reply by birkenfeld** [reply]
You start with `err`, which is the `PyErr::value()`. Then you can call `err.getattr("args")?.get_item(0)?`.

**Reply by bachue** [reply]
@birkenfeld It works! Thank you!
